internal/cmd/db_destroy.go

```diff
@@ -18,51 +21,85 @@ func init() {
 	Use:               "destroy database_name",
 	Short:             "Destroy a database.",
-	Args:              cobra.ExactArgs(1),
+	Args:              cobra.MinimumNArgs(1),
 	ValidArgsFunction: dbNameArg,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		cmd.SilenceUsage = true
+
 		client, err := createTursoClientFromAccessToken(true)
 		if err != nil {
 			return err
 		}
 
-		db, err := getDatabase(client, args[0])
-		if err != nil {
-			return nil
+		if len(args) > 1 {
+			return handleDestroyMultipleDBs(args, client)
 		}
 
-		name := args[0]
-		if instanceFlag != "" {
-			if db.Group != "" {
-				return fmt.Errorf("group databases do not support instance destruction.\nUse %s instead", internal.Emph("turso group locations rm "+name))
-			}
-			return destroyDatabaseInstance(client, name, instanceFlag)
-		}
+		return handleDestroySingleDB(args, client)
+	},
+}
 
-		if locationFlag != "" {
-			if db.Group != "" {
-				return fmt.Errorf("group databases do not support location destruction.\nUse %s instead", internal.Emph("turso group locations rm "+name+" "+locationFlag))
-			}
-			return destroyDatabaseRegion(client, name, locationFlag)
+func handleDestroySingleDB(args []string, client *turso.Client) error {
+	name := args[0]
+
+	db, err := getDatabase(client, name)
+	if err != nil {
+		return nil
+	}
+
+	if instanceFlag != "" {
+		if db.Group != "" {
+			return fmt.Errorf("group databases do not support instance destruction.\nUse %s instead", internal.Emph("turso group locations rm "+name))
 		}
+		return destroyDatabaseInstance(client, name, instanceFlag)
+	}
 
-		if yesFlag {
-			return destroyDatabase(client, name)
+	if locationFlag != "" {
+		if db.Group != "" {
+			return fmt.Errorf("group databases do not support location destruction.\nUse %s instead", internal.Emph("turso group locations rm "+name+" "+locationFlag))
 		}
+		return destroyDatabaseRegion(client, name, locationFlag)
+	}
 
-		fmt.Printf("Database %s, and all its data will be destroyed.\n", internal.Emph(name))
+	if yesFlag {
+		return destroyDatabases(client, args)
+	}
 
-		ok, err := promptConfirmation("Are you sure you want to do this?")
-		if err != nil {
-			return fmt.Errorf("could not get prompt confirmed by user: %w", err)
-		}
+	fmt.Printf("Database %s and all its data will be destroyed.\n", internal.Emph(name))
 
-		if !ok {
-			fmt.Println("Database destruction avoided.")
-			return nil
-		}
+	ok, err := promptConfirmation("Are you sure you want to do this?")
+	if err != nil {
+		return fmt.Errorf("could not get prompt confirmed by user: %w", err)
+	}
 
-		return destroyDatabase(client, name)
-	},
+	if !ok {
+		fmt.Println("Database destruction avoided.")
+		return nil
+	}
+
+	return destroyDatabases(client, args)
+}
+
+func handleDestroyMultipleDBs(args []string, client *turso.Client) error {
+	if instanceFlag != "" || locationFlag != "" {
+		return errors.New("can not use location nor instance flag when deleting more than 1 database")
+	}
+
```

Comment:
What happens when you try to destroy a DB that doesn't exist? Is the error message clear?
When destroying a single one we do `getDatabase`, so we exit early if the DB does not exist.

Reply:
Is this question related to when the instance or the location flag is passed or in general?

I ran `turso db destroy idk idk2 idk3`, where idk3 doesnt exist, and got `Error: database idk3 not found. List known databases using turso db list`, and idk1 and 2 got deleted, but this is "random", it was just a coincidence that idk3's  closure ended lastly.

If I ran it with only dbs that doesn't exit, one of the listed dbs will be used as an error "randomly".

I'm not sure how to handle each error individually using a waitgroup, because it always stop at the first error encountered. From what I understand we have 2 options:
1 - run the code linearly, and stop at the first error, printing the success msgs that happened before this error occurred, if any
2 - create a "custom concurrent error handler" and print all errors/success messages to the console

I think 2 would make more sense, I'm happy to work on it. What do you think?

Reply:
I think what you have right now is good enough!